appformer-kogito-bridge/src/main/java/org/appformer/kogito/bridge/client/keyboardshortcuts/KeyboardShortcutsServiceProducer.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.appformer.kogito.bridge.client.keyboardshortcuts;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+
+import elemental2.dom.DomGlobal;
+import org.appformer.kogito.bridge.client.interop.WindowRef;
+
+/**
+ * Produces {@link KeyboardShortcutsService} beans according to whether the envelope API is available or not
+ */
+public class KeyboardShortcutsServiceProducer {
+
+    @Produces
+    @ApplicationScoped
+    public KeyboardShortcutsApi produce() {
+
+        if (WindowRef.isEnvelopeAvailable()) {
+            return new KeyboardShortcutsService();
+        }
+
+        DomGlobal.console.info("[KeyboardShortcutsServiceProducer] Envelope API is not available. Producing NoOpKeyboardShortcutsService");
```

Comment:
Isn't this rather warning/error than info? My thoughts were all channels in kogito should have an envelope. Have just some of them the envelope?

Or is it that small subparts/panels/components of editor can have registered different shortcuts so it will be quite common that no envelope is present?

Reply:
Changed to 'debug'. The envelope is needed on all channels. This 'debug' message will be displayed on BC, since there's no envelope available there.